datashader/transfer_functions/_cuda_utils.py

```diff
@@ -75,32 +75,31 @@ def masked_clip_2d(data, mask, lower, upper):
 # Behaviour of numba.cuda.atomic.max/min changed in 0.50 so as to behave as per
 # np.nanmax/np.nanmin
 if LooseVersion(numba.__version__) >= LooseVersion("0.51.0"):
-    @cuda.jit
-    def cuda_atomic_nanmin(ary, idx, val)
+    @numba.jit
```

Comment:
Guess I'm missing something, what was wrong with `cuda.jit` here?

Reply:
if I use the `@cuda.jit` decorator, I get the following error:

```bash
TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Untyped global name 'cuda_atomic_nanmax': cannot determine Numba type of <class 'numba.cuda.compiler.Dispatcher'>
```
and since calling a numba.jit function from CUDA automatically compiles an equivalent CUDA device function.


Although going through some numba examples, I realized that `@cuda.jit(device=True)` would achieve the same device function, and the result works. I have updated the code. Thanks for pointing this out

Reply:
Thanks for looking into it, really appreciate it!